vernac/vernacentries.ml

```diff
@@ -2082,9 +2082,21 @@ let vernac_check_guard ~pstate =
       (str ("Condition violated: ") ++s)
   in message
 
+let rec holds = let open Attributes in function
+  | Var { name; value = None } -> Option.has_some @@ Sys.getenv_opt name
+  | Var { name; value = Some v } -> Sys.getenv_opt name = Some v
+  | Or(a,b) -> holds a || holds b
+  | And(a,b) -> holds a && holds b
+  | Not a -> not (holds a)
+
 (* We interpret vernacular commands to a DSL that specifies their
    allowed actions on proof states *)
-let translate_vernac ~atts v = let open Vernacextend in match v with
+let translate_vernac ~atts v =
```

Comment:
I would rather change Vernacinterp.interp_control
For instance the current change makes `#[if(FALSE)] Fail Check nat.` fail where I would expect it to succeed by being a noop.

Reply:
Oh right. I did put the thing where attributes are usually handled. Maybe it is still the right place, but wiping out all controls.